lib/src/native/realm_core.dart

```diff
@@ -151,6 +151,9 @@ class _RealmCore {
       if (config.fifoFilesFallbackPath != null) {
         _realmLib.realm_config_set_fifo_path(configHandle._pointer, config.fifoFilesFallbackPath!.toUtf8Ptr(arena));
       }
+
+      _realmLib.realm_config_set_disable_format_upgrade(configHandle._pointer, config.disableFormatUpgrade);
```

Comment:
This should be called only if the flag is set to true

Reply:
I wondered whether to do it because the similar checks were deleted for the rest of properties that have default value `false`. For example isInMemory and isReadOnly.

Reply:
hm, I am not sure why